### PR TITLE
[Xthor] Improved research (limited to 96 results instead of 32 before) resolves #6392

### LINKED ISSUE #6392: Search incomplete on Xthor
Hi, I have a strange problem going on since several weeks now, the search on Xthor, doesn't give me back the full answer, for example, if I give a search on " The Walking dead", on Xthor I only receive matches for French or Multi.French, and only from the lastest 15 days. Of course a search on Xthor directly, give me all the episode available.
I dont have the problem on other trackers.

Jackett version : 0.12.1007.0

### PR DESCRIPTION
**Pull request by Tawagot0**
Resolves #6392
firefox_E61EPtEvHE

### CODE REVIEW

**Comment by Tawagot0** [comment]
Idea to implement for later: add an option in the config page to change the maximum number of pages (with a limit of course)

File: src/Jackett.Common/Indexers/Xthor.cs [hunk 1]
```diff
@@ -204,74 +211,85 @@ protected override async Task<IEnumerable<ReleaseInfo>> PerformQuery(TorznabQuer
...
+            for (var page = 0; page <= MaxSearchPageLimit; page++)
             {
-                // Deserialize our Json Response
-                var xthorResponse = JsonConvert.DeserializeObject<XthorResponse>(results);
+                var request = BuildQuery(searchTerm, query, ApiEndpoint);
 
-                // Check Tracker's State
-                CheckApiState(xthorResponse.error);
+                // Add option page at the end of the request variable
+                request = request + "&page=" + page;
```
Comment (ngosang) [inline src/Jackett.Common/Indexers/Xthor.cs:220]: change `BuildQuery` function to include the page and do all the logic to generate the URL in there.

File: src/Jackett.Common/Indexers/Xthor.cs [hunk 2]
```diff
@@ -204,74 +211,85 @@ protected override async Task<IEnumerable<ReleaseInfo>> PerformQuery(TorznabQuer
...
+                                UploadVolumeFactor = 1,
+                                DownloadVolumeFactor = (torrent.freeleech == 1 ? 0 : 1),
+                                Guid = guid,
+                                Comments = comments,
+                                Link = link,
+                                TMDb = torrent.tmdb_id
+                            };
+
+                            //TODO make consistent with other trackers
+                            if (DevMode)
```
Comment (ngosang) [inline src/Jackett.Common/Indexers/Xthor.cs:276]: remove all dev code

File: src/Jackett.Common/Indexers/Xthor.cs [hunk 3]
```diff
@@ -204,74 +211,85 @@ protected override async Task<IEnumerable<ReleaseInfo>> PerformQuery(TorznabQuer
...
+                            }
+
+                            // issue #8759 replace vostfr and subfrench with English
+                            if (ConfigData.Vostfr.Value)
+                                torrent.name = torrent.name.Replace("VOSTFR", "ENGLISH").Replace("SUBFRENCH", "ENGLISH");
+
+                            var publishDate = DateTimeUtil.UnixTimestampToDateTime(torrent.added);
+                            //TODO replace with download link?
+                            var guid = new Uri(TorrentDescriptionUrl.Replace("{id}", torrent.id.ToString()));
+                            var comments = new Uri(TorrentCommentUrl.Replace("{id}", torrent.id.ToString()));
```
Comment (ngosang) [inline src/Jackett.Common/Indexers/Xthor.cs:252]: Change this code for me:
```
        private string TorrentCommentUrl => TorrentDescriptionUrl;
        private string TorrentDescriptionUrl => SiteLink + "details.php?id={id}";
```
to
```
private string TorrentCommentUrl => SiteLink + "details.php?id=";
```

and
```
                            //TODO replace with download link?
                            var guid = new Uri(TorrentDescriptionUrl.Replace("{id}", torrent.id.ToString()));
                            var comments = new Uri(TorrentCommentUrl.Replace("{id}", torrent.id.ToString()));
```
to
```
var comments = new Uri(TorrentCommentUrl + id);
```

and

```
 Guid = guid,
 Comments = comments,
```
to
```
 Guid = comments,
 Comments = comments,
```

remember to test the details link is working after this change

**Comment by Tawagot0** [comment]
Bug found: some results appear double / triple (probably due to the loop)
Surely need help to fix this

**Comment by Tawagot0** [comment]
**UPDATE** 
_Bug found_: some results appear double / triple (probably due to the loop)
Surely need help to fix this

File: src/Jackett.Common/Indexers/Xthor.cs (hunk 2 above)
Comment (ngosang) [reply src/Jackett.Common/Indexers/Xthor.cs:276]: update: don't change dev code

**Comment by ngosang** [comment]
> Bug found: some results appear double / triple (probably due to the loop)

This can be fixed removing duplicates, but I think you should find if the page=0 and page=1 are returning the same results. are you sure page=0 is the first page?

**Comment by Tawagot0** [comment]
> 
> 
> > Bug found: some results appear double / triple (probably due to the loop)
> 
> This can be fixed removing duplicates, but I think you should find if the page=0 and page=1 are returning the same results. are you sure page=0 is the first page?

Yes that's right i just changed it, however this is a problem with the loop, if i increase the variable: `MaxSearchPageLimit` for example to 5, i will have 5 times the same result.
I will try to see with breakpoints.

firefox_gNgEGi3NN8

**Comment by ngosang** [comment]
This indexer is a bit weird. It has a cache, you can try to comment that code to test => https://github.com/Jackett/Jackett/blob/c194aadbe8fd07af4482f9a90c9e532f5ab4014e/src/Jackett.Common/Indexers/Xthor.cs#L423

**Comment by ngosang** [comment]
Since I don't have an account on this site I can't provide more help.
If you want to lend me the account send me an email.

**Comment by JigSawFr** [comment]
Implemented in another PR.
But API seems to be buggy.

**Pull request closed without merging**